### DISCUSSION TITLE
S-curve and baud rate adjustments

### DISCUSSION THREAD

**Discussion by darco415** [Jyers/Marlin#29] [Ideas]
Ive noticed other fw have implemented s_curve as well as serial port speed to 250000, Curious if this fw will also implement,
thxs

**Comment by Jyers** [comment]
I have not heard of s_curve, is that something that can be enabled from the configuration files? Also I wasn't aware the serial port on the E3V2 was capable of higher port speeds, but that would be an easy change if it is possible.

**Reply by darco415** [reply]
Yes s_curve is there in config.h file but I dont think its  simply uncommenting it will make it work (?). Also I have seen  serial /250000 in Smith3D fw so Im assuming its working there somehow.

**Reply by Jyers** [reply]
Interesting, I will definitely give the baud rate a try and look into the s_curve setting!

**Reply by Benetos** [reply]
Yeah it is under config. I am running it on my own build of the previous ui you released.

**Reply by Benetos** [reply]
Just a update but I am running it currently as well with no issue.

**Reply by jjfs127** [reply]
So it's just a matter of uncommentting: `#define S_CURVE_ACCELERATION` and setting `#define BAUDRATE 250000`? Thanks

**Reply by t3chguy** [reply]
If you care more about stability than raw speed I would suggest using https://github.com/scottmudge/OctoPrint-MeatPack/ which keeps your baudrate but packs more gcode per word. By not raising your baud rate you'll have less potential interference issues

**Reply by jjfs127** [reply]
Is MeatPack support already integrated or needs config setting changes? Basically simply install the plugin?

**Reply by t3chguy** [reply]
MeatPack support is in latest bugfix on Marlin, needs to be enabled using `#define MEATPACK`

**Reply by jjfs127** [reply]
Sorry, would that be an existing line needing uncommentting? I don't see it in either config file.

**Reply by t3chguy** [reply]
https://github.com/MarlinFirmware/Configurations/blob/import-2.0.x/config/default/Configuration_adv.h#L3325 yes but its probably not in the config files in this repo yet.

**Reply by jjfs127** [reply]
After install the Meatpack plugin and putting in the line `#define MEATPACK`

I see and Octoprint can't connect:
```
Changing monitoring state from "Offline" to "Opening serial connection"
Changing monitoring state from "Opening serial connection" to "Connecting"
Connected to: PackingSerial<id=0xb34a4ef0, open=True>(port='/dev/ttyUSB0', baudrate=115200, bytesize=8, parity='N', stopbits=1, timeout=10.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor
Send: N0 M110 N0*125
Send: N0 M110 N0*125
No answer from the printer within the connection timeout, trying another hello
Send: N0 M110 N0*125
There was a timeout while trying to connect to the printer
Changing monitoring state from "Connecting" to "Offline"
Connection closed, closing down monitor
Changing monitoring state from "Offline" to "Opening serial connection"
```

**Reply by t3chguy** [reply]
I had the same then restarted the printer and it has been fine since.

**Reply by jjfs127** [reply]
Hmm, odd not going away.

**Reply by t3chguy** [reply]
I pulled in latest bugfix, not sure how old the bugfix in this fork is

**Reply by jjfs127** [reply]
Seems like that could the problem: https://github.com/scottmudge/OctoPrint-MeatPack/issues/12
Not sure if I know how to pull in the latest bugfix...

**Reply by arthurgeek** [reply]
I'm also unable to connect via Octoprint when the plugin is enabled. I'm using latest source from this repo which should contain the fix mentioned in https://github.com/scottmudge/OctoPrint-MeatPack/issues/12. Anyone fixed this?

**Reply by jjfs127** [reply]
Seems fine on my end after Jyers recent bugfix pull...but using the default plugin via the Octoprint plugins menu.

**Reply by darco415** [reply]
> If you care more about stability than raw speed I would suggest using https://github.com/scottmudge/OctoPrint-MeatPack/ which keeps your baudrate but packs more gcode per word. By not raising your baud rate you'll have less potential interference issues

is this similar to the arc-welder plugin? It also compesses gcode

**Reply by t3chguy** [reply]
@darco415 very different, this is entirely lossless. It just encodes the exact same GCode in a more efficient way because most GCode commands use very few unique characters. Arc Welder can yield higher compression but is lossy and more expensive to compute on both ends.

**Reply by darco415** [reply]
> @darco415 very different, this is entirely lossless. It just encodes the exact same GCode in a more efficient way because most GCode commands use very few unique characters. Arc Welder can yield higher compression but is lossy and more expensive to compute on both ends.

Ok thxs...I just uninstalled arc-welder and now meatpack is running....testing now

**Reply by t3chguy** [reply]
In theory there is nothing to stop you using both, Meatpack would losslessly compress the GCode output from Arc Welder which is just arc commands.

**Reply by darco415** [reply]
even better..thxs!

**Reply by Schnabulation** [reply]
Unrelated: has anyone tested arc-welder with our E3V2? As far as I heard it should improve the print quality quite a bit...

**Reply by Jyers** [reply]
I've been using the cura plugin for a bit. Haven't noticed any real improvement, but then again I haven't been doing a ton of real printing...

**Reply by jjfs127** [reply]
What's required on the Marlin side to set up arc-welder?

**Reply by t3chguy** [reply]
`#define ARC_SUPPORT` which I believe is the default.

**Reply by jjfs127** [reply]
> `#define ARC_SUPPORT` which I believe is the default.

Ah, you're right, that's the default.
`Recv: Cap:ARCS:1`

**Reply by darco415** [reply]
> In theory there is nothing to stop you using both, Meatpack would losslessly compress the GCode output from Arc Welder which is just arc commands.

so previously running w/o meatpack and only arc-welder (s_curve enabled), my 22hr print came out beautifully. Then I enabled meatpack w/arc-weld enabled. Same gcode, the print came out good but I noticed alot of internal stringing. I then disabled arc-welder, ran a benchy, it came out beautiful with some very faint stringing.....Ill leave arc-welder disable for the meantime, see how things go

**Reply by rorywquin** [reply]
Yea - I tried a print with and without ArcWelder and couldn't really see and improvement in the print. I think it probably comes into it's own with larger prints with lots of curves. Installed MeatPack yesterday and it kept crashing the serial connection - Octoprint gave a message about development software causing instability.

**Comment by Benetos** [comment]
I don’t know if I have it enabled but if not I am going to. I know the
plug-in 3.5.1 in cura was supposedly given some issues but it might be
fixed.

The arc welding probably won’t make a huge difference unless it’s a larger
print with more than one or two curves. Something like a curved container
or a vertical wave or the Ford logo.
